### ISSUE TITLE
`outputSelection` with contract name and not `*` leads to empty compiler output in `<=0.5.8`

### ISSUE THREAD

**Issue by kuzdogan** [ethereum/solidity#14214]
### Summary:

Providing a contract name in the contract selection of the `outputSelection` field of the JSON input is outputting empty JSON fields in versions `<=0.5.8`.

Doesn't work:
```json
"outputSelection": {
  "*": {
    "Storage": ["*"]
  }
}
```

Works:
```json
"outputSelection": {
  "*": {
    "*": ["*"]
  }
}
```

### Details

I've encountered multiple issues about the `outputSelection`. 

- Initially https://github.com/ethereum/solidity/issues/13985 about the **file selection** in the output selection in `v0.6.5`.
- Recently I've come across another issue https://github.com/ethereum/solc-js/issues/690 in `<=0.4.19`
  - I thought it happens because of the introduction of the `outputSelection` in that version.
  - Shouldn't be related to solc as the field wasn't introduced back then and unfortunately there was no valid field check.
  - Hence issue opened in solc-js to handle the JSON translation for earlier versions

Digging into the issue https://github.com/ethereum/sourcify/issues/1017 in Sourcify, I came across this issue which affects versions `<=0.5.8` but not `>=0.5.9`. I couldn't find any related changes in this version about the `outputSelection` so I'm opening the issue.

To reproduce, here's the compiler input JSON:
```json
{
  "language": "Solidity",
  "settings": {
    "optimizer": { "enabled": true, "runs": 200 },
    "outputSelection": {
      "*": {
        "Storage": ["evm.bytecode"]
      }
    }
  },
  "sources": {
    "Storage.sol": {
      "content": "pragma solidity >=0.4.0 <0.9.0;\n\ncontract Storage {\n\n    uint256 number;\n\n    function store(uint256 num) public {\n        number = num;\n    }\n\n    function retrieve() public view returns (uint256){\n        return number;\n    }\n}\n"
    }
  }
}
```

This would generate the following unexpected output:
```json
{
  "contracts": {
    "Storage.sol": {
      "Storage": {
        "evm": {
          "bytecode": {
            "linkReferences": {},
            "object": "",
            "opcodes": "",
            "sourceMap": ""
          }
        }
      }
    }
  },
  "sources": {
    "Storage.sol": {
      "id": 0
    }
  }
}
```

Changing the `"Storage"` field in the JSON to `"*"` will lead to an expected compiler output with bytecode.

The contract is fairly simple and attached here: 
[Storage.sol.txt](https://github.com/ethereum/solidity/files/11452181/Storage.sol.txt)

## Environment

- Compiler version: affected `<=0.5.8`
- Target EVM version (as per compiler settings): `default`
- Framework/IDE (e.g. Truffle or Remix): Both in `0.5.8+commit.23d335f2.Darwin.appleclang` downloaded with [solc-select](https://github.com/crytic/solc-select). I was able to reproduce in Remix, as well as in Sourcify, which uses solc-js on Mac ARM.
- Operating system: MacOS ARM 64bit

**Comment by ekpyron** [comment]
Since we won't touch old releases, the fix for this will probably be something similar to https://github.com/ethereum/solc-js/issues/690, although we may want to come up with a nicer cross-platform way to do such input/output translations.

**Comment by kuzdogan** [comment]
Right. Putting this for the record and if it is indeed a bug, maybe to be listed in bugs. 

We just went forward with the workaround of using `*` on all levels, so all fine if this should be closed.

**Comment by cameel** [comment]
Looks like the bug was fixed by #6606, but it's not in the changelog.

We can't do anything more about it directly in solc so this was rightly closed, but IMO it's still useful to keep the list of such possible workarounds in solc-js repo. Maybe we'll eventually find some time to deal with them. I reported it as https://github.com/ethereum/solc-js/issues/755.
